swagger.json

```diff
@@ -0,0 +1,421 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "description": "banidb-api docs",
+    "version": "1",
+    "title": "banidb-api"
+  },
+  "servers": [
+    {
+      "url": "",
+      "description": "live endpoint"
+    }
+  ],
+  "paths": {
+    "/health": {
+      "get": {
+        "summary": "Health check",
+        "description": "Make sure api is up and running",
+        "responses": {
+          "200": {
+            "description": "successful operation"
+          }
+        }
+      }
+    },
+    "/search/{query}": {
+      "get": {
+        "summary": "Search",
+        "description": "Enter a search query and get lines that match.",
+        "parameters": [
+          {
+            "name": "query",
+            "in": "path",
+            "description": "Search query",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "searchtype",
+            "in": "query",
+            "description": "Type of search. { 0: First letter each word from start (Gurmukhi), 1: First letter each word anywhere(Gurmukhi), 2: Full Word (Gurmukhi), 3: Full Word Translation (English), 4: Romanized Gurmukhi (English), 5: Ang, 6: Main Letter (Gurmukhi), 7: Romanized first letter anywhere (English) }",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/SeaerchType"
+            }
+          },
+          {
+            "name": "source",
+            "in": "query",
+            "description": "Source to search in. { G: Guru Granth Sahib Ji, D: Dasam Granth Sahib, B: Bhai Gurdas Ji Vaaran, A: Amrit Keertan, S: Bhai Gurdas Singh Ji Vaaran N: Bhai Nand Lal Ji Vaaran, R: Rehatnamas & Panthic Sources }",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/Source"
+            }
+          },
+          {
+            "name": "ang",
+            "in": "query",
+            "description": "Which ang to return. Required if searchtype is 5",
+            "required": false,
+            "type": "number"
+          },
+          {
+            "name": "raag",
+            "in": "query",
+            "description": "Which raag to limit search to.",
+            "required": false,
+            "type": "number"
+          },
+          {
+            "name": "writer",
+            "in": "query",
+            "description": "Which writer to limit search to.",
+            "required": false,
+            "type": "number"
+          },
+          {
+            "name": "page",
+            "in": "query",
+            "description": "Pagination param. Which page to return.",
```

Comment:
Though "Pagination param" is specified, there could be addition clarity with "Which results page to return"